packages/components/tree-v2/src/hooks/useCheck.ts

```diff
@@ -0,0 +1,220 @@
+import { nextTick, ref, watch } from 'vue'
+import { NODE_CHECK_CHANGE, NODE_CHECK } from '../emits'
+import type { Ref } from 'vue'
+import type {
+  ITreeProps,
+  TreeKey,
+  TreeNode,
+  Tree,
+  TreeNodeData,
+} from '../tree.type'
+
+const DELETE = 'delete'
+const ADD = 'add'
+
+export function useCheck(props: ITreeProps, tree: Ref<Tree>, emit) {
+  const checkedKeys = ref<Set<TreeKey>>(new Set())
+  const indeterminateKeys = ref<Set<TreeKey>>(new Set())
```

Comment:
**Question:** Do we really need these to be reactive? I didn't see any usage of reactive effect. cc @sxzz WDYT

Reply:
Emmm, maybe we can use it without `ref`? I am not very familiar with this part.
```suggestion
  const checkedKeys = new Set<TreeKey>()
  const indeterminateKeys = new Set<TreeKey>()
```

Reply:
I can't do that, i need to determine the check status of the node